spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /users/:user_id/posts' do
+    it 'works! (now write some real specs)' do
+      get '/users/1/posts'
+      expect(response).to have_http_status(200)
+      expect(response).to render_template('posts/index')
+      expect(response.body).to include('Posts from user 1')
+    end
+  end
+
+  describe 'GET /users/:user_id/posts/:id' do
+    it 'works! (now write some real specs)' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(200)
+      expect(response).to render_template('posts/show')
+      expect(response.body).to include('Post 1 from User 1')
+    end
+  end
+end
```

Comment:
- [ ] Great job writing tests for your controllers. However, at this moment, the tests are failing. Kindly look into them and fix them.

### Posts errors
![post-error](https://github.com/IsmailChangezi/blog-app-new/assets/26597432/827ca784-8c84-4c74-9821-36f0dce58c7c)

![post-error-1](https://github.com/IsmailChangezi/blog-app-new/assets/26597432/6ca8406f-e15e-4b37-a3c3-437c521201c9)